packages/safe-tools-client/app/components/setup-safe-modal/index.gts

```diff
@@ -12,22 +17,32 @@ interface Signature {
   Element: HTMLElement;
   Args: { 
     isOpen: boolean 
-    loading?: boolean;
     onClose: () => void;
   };
 }
 
-//TODO: replace with correct flags and logic
-const gasCost = `0.001899365 ETH (USD$3.01)`;
-const hasEnoughGas = true;
-
 export default class SetupSafeModal extends Component<Signature> {
+  //TODO: replace with correct flags and logic
+  gasCost = `0.001899365 ETH (USD$3.01)`;
+  notEnoughBalance = false;
+
+  @tracked provisioning = false;
+
+  @action async createSafe(): Promise<void> { 
+     //TODO: use sdk method createSafeWithModuleAndGuard
+    this.provisioning = true;
+    setTimeout(() => {
+      console.log('Create safe')
+      this.provisioning = false;
+      this.args.onClose();
+    }, 2000); 
+  }
```

Comment:
This pattern of toggling a flag to indicate when something is in progress is one reason to use an [Ember Concurrency task](http://ember-concurrency.com/docs/tutorial/refactor) as `createSafe.isRunning` would be taken care of automatically. (You could also then use `timeout` for the currently-fake timer.)

Reply:
Provisioning will require a signing request, so I think we will want to have some other states in the action chin during this similar to what we do in web-client.

Reply:
got it, will take into consideration once the integration starts